### PR TITLE
[v4.1]Raven db 11233 4.1

### PR DESCRIPTION
**Pull request by maximburyak**

### CODE REVIEW

File: src/Raven.Client/Documents/Changes/DatabaseChanges.cs [hunk 1]
```diff
@@ -343,17 +343,14 @@ private DatabaseConnectionState GetOrAddConnectionState(string name, string watc
             catch (OperationCanceledException e)
             {
                 NotifyAboutError(e);
+                _tcs.TrySetException(e);
```
Comment (ayende) [inline src/Raven.Client/Documents/Changes/DatabaseChanges.cs:346]: Shouldn't this be `TrySetCancelled`?

**Comment by ayende** [comment]
I'm not sure how this is solving the race condition on the `OnError` method?

**Comment by ppekrol** [comment]
It covers an issue where the exception happenes during the first initialization of the Changes API. I think this should be enough. We cannot have a consistent behavior when `OnError` would be passed to the `Changes()` method, due to caching of the `DatabaseChanges`. Which means that second call to that method would be racy again.

File: src/Raven.Client/Documents/Changes/DatabaseChanges.cs (hunk 1 above)
Comment (maximburyak) [reply src/Raven.Client/Documents/Changes/DatabaseChanges.cs:346]: I'm not sure, It's possible that we want to track the exception stack trace, or we just want to know that it was canceled?

Comment (ppekrol) [reply src/Raven.Client/Documents/Changes/DatabaseChanges.cs:346]: I think we should go with `TrySetCanceled`. You will not get much more details here anyway.

**Pull request merged**